### PR TITLE
solana v1.18.18 - including solana-test-validator

### PR DESCRIPTION
**Pull request by DomBlack**
This commit adds `solana-test-validator` to the list of binaries installed by the Solana formula, [which is documented as coming built-in](https://solana.com/developers/guides/getstarted/setup-local-development#5-setup-a-localhost-blockchain-cluster).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by DomBlack** [comment]
Could somebody trigger the `CI / macOS 12-x86_64` run again please. The truncated error reports;

```
Error: Installed package `solana-stake-accounts v1.18.18 (/private/tmp/solana-20240716-8315-6uhnem/solana-1.18.18/stake-accounts)` (executable `solana-stake-accounts`)
```

However that binary built ok and I can't see any errors the full output logs (maybe I'm blind though)

```
Compiling solana-stake-accounts v1.18.18 (/private/tmp/solana-20240716-8315-6uhnem/solana-1.18.18/stake-accounts)
      Finished `release` profile [optimized] target(s) in 12m 01s
    Installing /usr/local/Cellar/solana/1.18.18/bin/solana-stake-accounts
     Installed package `solana-stake-accounts v1.18.18 (/private/tmp/solana-20240716-8315-6uhnem/solana-1.18.18/stake-accounts)` (executable `solana-stake-accounts`)
```

Given it passed on the other machines, I'm wondering if it hit some sort of timeout issue?

File: Formula/s/solana.rb
Comment (SMillerDev) [inline Formula/s/solana.rb:43]: Those are some pretty heavy dependencies to pull in. Is this a very popular feature that you're proposing here?

Comment (DomBlack) [reply Formula/s/solana.rb:43]: I believe so, if you are developing against Solana the [docs](https://docs.solanalabs.com/cli/examples/test-validator) and [local dev guide](https://solana.com/developers/guides/getstarted/setup-local-development) both use the test validator and state it comes with the CLI, the install instructions for getting the CLI [call out installing via brew](https://docs.solanalabs.com/cli/install#use-homebrew) which I found confusing as it didn't come with the test validator as expected.

Comment (DomBlack) [reply Formula/s/solana.rb:43]: Sorry for the bump @bevanjkay / @SMillerDev - would it be possible to get an approval or further the discussion on this please, I'm going to become blocked on this and may need to create my own tap'ed version to unblock myself.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @DomBlack!

**Pull request merged**
